Expose aggressive_recomputation as an inductor config

Yuzhen11 opened the pull request:
Summary:
As title.

We found aggressive_recomputation shows memory savings (7% on APS COFFEE model) with 2% QPS loss.

It also gives very promising signal on our auto ac experiments: https://docs.google.com/document/d/1S2qgMg1CwAQ4U1Ffuk2epbEOx06ogZhioX2jKCwL7ZQ/edit

 {F1426175073}

Test Plan:
APS COFFEE from silverlakeli
- Zoom of baseline job: https://www.internalfb.com/intern/zoomer/?profiling_run_fbid=927380488801910&tab=overview
- Zoom of job with aggressive_recomputation: https://www.internalfb.com/intern/zoomer/?profiling_run_fbid=1126815608217470&tab=overview


APS 1100x shrunk version:
- baseline: https://www.internalfb.com/mast/job/aps-yuzhenhuang-afe049505a
- test: https://www.internalfb.com/mast/job/aps-yuzhenhuang-709e41bf0d
Memory from 42.98% -> 41.04%.

Reviewed By: yf225, yuxihu, silverlakeli, richqyz

Differential Revision: D53248057




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

yanboliang reviewed (commented):
I think if it's ready to expose as a public config, it should be part of [functorch config](https://github.com/pytorch/pytorch/blob/main/torch/_functorch/config.py) rather than inductor config.

cc @anijain2305 on AC readiness.

Yuzhen11 commented:
> I think if it's ready to expose as a public config, it should be part of [functorch config](https://github.com/pytorch/pytorch/blob/main/torch/_functorch/config.py) rather than inductor config.
> 
> cc @anijain2305 on AC readiness.

Good point. Let me update it.

Yuzhen11 committed:
Expose aggressive_recomputation as an inductor config (#118943)

Summary:

As title.

We found aggressive_recomputation shows memory savings.

Reviewed By: yf225, yuxihu, silverlakeli, richqyz

Differential Revision: D53248057
